thunder/torch/__init__.py

```diff
@@ -566,7 +566,7 @@ def full(
     shape: Sequence[int], fill_value: NumberLike, *, device: None | DeviceLike = None, dtype: None | dtypeLike = None
 ) -> TensorLike:
     device = to_device(maybe_get_default_device(device))
-    dtype = to_dtype(maybe_get_default_dtype(dtype))
+    dtype = to_dtype(dtype)
```

Comment:
Looking at `torch.full` implementation 
https://github.com/pytorch/pytorch/blob/cd307fb0b1a833f9297d2233653b514ed4aa3163/aten/src/ATen/native/TensorFactories.cpp#L584-L604

When dtype is not passed explicitly, dtype is determined by
```python
if fill_value == bool:
   dtype = bool
elif fill_value.is_integer:
   dtype = LongInt
elif fill_value.is_complex:
   dtype = default_complex_dtype # complex type corresponding to default dtype
else: # floating
   dtype = default_floating_dtype
```

So, I think we need to get dtype from default torch dtype if fill_value is floating or complex.

Reply:
More precisely, we need to get the default floating point dtype or default complex dtype if dtype is not specified and the inputing is a floating point or complex number (respectively)